cdk/dcm_floodlight/transform.js

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable no-param-reassign */
+const get = require("get-value");
+const {
+  getValueFromMessage,
+  removeUndefinedAndNullValues,
+  isDefinedAndNotNull
+} = require("rudder-transformer-cdk/build/utils");
+const { CustomError, getIntegrationsObj, isEmpty } = require("../../v0/util");
+
+// append properties to endpoint
+// eg: ${endpoint}key1=value1;key2=value2;....
+function appendProperties(endpoint, payload) {
+  Object.keys(payload).forEach(key => {
+    endpoint += `${key}=${payload[key]};`;
+  });
+
+  return endpoint;
+}
+
+// transform webapp dynamicForm custom floodlight variable
+// into {u1: value, u2: value, ...}
+function transformCustomVariable(customFloodlightVariable) {
+  const customVariable = {};
+  customFloodlightVariable.forEach(item => {
+    if (!isEmpty(item.from) && !isEmpty(item.to)) {
+      // remove u if already there
+      customVariable[
+        `u${item.from.trim().replace(/u/g, "")}`
+      ] = `[${item.to.trim()}]`;
+    }
+  });
+
+  return customVariable;
+}
+
+// valid flag should be provided [1|true] or [0|false]
+function isValidFlag(key, value) {
+  if (["true", "1"].includes(value.toString())) {
+    return 1;
+  }
+  if (["false", "0"].includes(value.toString())) {
+    return 0;
+  }
+
+  throw new CustomError(
+    `[DCM Floodlight]:: ${key}: valid parameters are [1|true] or [0|false]`,
+    400
+  );
+}
+
+function trackPostMapper(input, mappedPayload, rudderContext) {
+  const { message, destination } = input;
+  const { advertiserId, conversionEvents } = destination.Config;
+  let { activityTag, groupTag } = destination.Config;
+
+  const baseEndpoint = "https://ad.doubleclick.net/ddm/activity/";
+
+  let event = getValueFromMessage(message, "event");
+  if (!event) {
```

Comment:
```suggestion
  if (!isEmpty(event)) {
```

Reply:
`getValueFromMessage()` takes care of this